### PR TITLE
move a few more common crates to 'crates/' directory

### PR DESCRIPTION
**Pull request by bmwill**

### CODE REVIEW

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M bounded-executor 0.1.0 (workspace, path 'crates/bounded-executor')
    * source changed from path 'common/bounded-executor'
    * (unchanged features: [none])
  M diem-bitvec 0.1.0 (workspace, path 'crates/diem-bitvec')
    * source changed from path 'common/bitvec'
    * (unchanged features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M bounded-executor 0.1.0 (workspace, path 'crates/bounded-executor')
    * source changed from path 'common/bounded-executor'
    * (unchanged features: [none])
  M diem-bitvec 0.1.0 (workspace, path 'crates/diem-bitvec')
    * source changed from path 'common/bitvec'
    * (unchanged features: [none])
```

File: Cargo.toml [hunk 1]
```diff
@@ -25,7 +23,6 @@ members = [
     "common/short-hex-str",
     "common/temppath",
     "common/time-service",
-    "common/workspace-hack",
```
Comment (gregnazario) [inline Cargo.toml:28]: I really wish we had a better name for this than `workspace-hack`

**Review by gregnazario — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (bmwill) [reply Cargo.toml:28]: We'll we didn't' really name it, its something that was named by the rust lang project

**Pull request closed without merging**